skyportal/handlers/api/followup_request.py

```diff
@@ -433,6 +440,15 @@ def get(self, followup_request_id=None):
             followup_requests = followup_requests.join(
                 obj_subquery, FollowupRequest.obj_id == obj_subquery.c.id
             )
+        if instrumentID:
+            allocation_query = Allocation.query_records_accessible_by(
+                self.current_user
+            ).filter(Allocation.instrument_id == instrumentID)
+            allocation_subquery = allocation_query.subquery()
+            followup_requests = followup_requests.join(
+                allocation_subquery,
+                FollowupRequest.allocation_id == allocation_subquery.c.id,
+            )
```

Comment:
This seems more complicated than just getting the instrument and all associated followup requests. Can you add a comment explaining why we need the allocations subquery?

Reply:
yeah..